src/main/cms/cms.c

```diff
@@ -801,199 +683,170 @@ STATIC_UNIT_TESTED uint16_t cmsHandleKey(displayPort_t *pDisplay, uint8_t key)
             currentCtx.cursorRow = 0;    // Goto top in any case
         }
     }
-
     if (key == KEY_UP) {
         currentCtx.cursorRow--;
-
         // Skip non-title labels
         if ((pageTop + currentCtx.cursorRow)->type == OME_Label && currentCtx.cursorRow > 0)
             currentCtx.cursorRow--;
-
         if (currentCtx.cursorRow == -1 || (pageTop + currentCtx.cursorRow)->type == OME_Label) {
             // Goto previous page
             cmsPagePrev(pDisplay);
             currentCtx.cursorRow = pageMaxRow;
         }
     }
-
     if (key == KEY_DOWN || key == KEY_UP)
         return res;
-
     p = pageTop + currentCtx.cursorRow;
-
     switch (p->type) {
-        case OME_Submenu:
+    case OME_Submenu:
+        if (key == KEY_RIGHT) {
+            cmsMenuChange(pDisplay, p->data);
+            res = BUTTON_PAUSE;
+        }
+        break;
+    case OME_Funcall:
+        ;
+        long retval;
```

Comment:
Codacy found an issue: [The scope of the variable 'retval' can be reduced.](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=6307959)